homeassistant/util/pil.py

```diff
@@ -2,9 +2,18 @@
 
 Can only be used by integrations that have pillow in their requirements.
 """
-from typing import Tuple
+import io
+from typing import Tuple, Union
 
-from PIL import ImageDraw
+from PIL import Image, ImageDraw, UnidentifiedImageError
+
+
+def convert_to_pil_image(image: bytes) -> Union[Image.Image, None]:
+    """Convert a camera image to a PIL Image."""
+    try:
+        return Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+    except UnidentifiedImageError:
```

Comment:
One thing that is weird here, is that now we are not giving any feedback to the user. Something just didn't work. However, this function cannot know what to report, as it doesn't know if it's an upload or how to refer to the image in a way that the user understands. 

I don't think that having this function is beneficial as we're promoting bad practice of swallowing errors.

Reply:
Perhaps it would be better to implement this functionality as a `validate_image_data` method on the `Image_processing` class, and log the error? I think this is better than having uncaught exceptions as this issue https://github.com/home-assistant/home-assistant/issues/31610

Reply:
I agree. Let's have Tensorflow catch it and report on it.

Reply:
OK added

Reply:
I'm honestly not sure if this function is _really_ helping at all. It's saving 1 line and it's less clear what is going on ? I think that we should just put this `try…except` in doods and TF.

Reply:
I think we should trow our own exception and use `try...except` instead the `if not image:`.

Reply:
It is not saving much code at this time but it will standardise the way PIL is used, and avoid the same issues coming up time and again if used. But I am open to guidance on the best way forward. Re throwing an HA exception, is there a suitable one, and do we not prefer logs to exceptions?

Reply:
I think that the PIL exception is fine, we should just call PIL directly in those integrations and don't have a helper function.

Reply:
To clarify, we want to catch the UnidentifiedImageError exception and log it, or are we going to allow it to just surface with no try.. except..?

Reply:
we are going to catch it inside `doods` and `tensorflow`. Copy paste the content of this function and paste it into `doods/__init__.py` etc.